### DISCUSSION TITLE
Fully Enclosed Shell: Free Download

### DISCUSSION THREAD

**Discussion by KytorIndustries** [sanni/cartreader#723] [General Chat]
I have designed a fully enclosed shell. It is a three part print which provides better protection and a nice looking fit and finish. I hope you enjoy.

Download here: https://www.printables.com/model/409926

Licensed CC BY 4.0, commercial use allowed with attribution.

**Comment by breyell** [comment]
Came here to promote this and glad to see you've already done that. Amazing work! Looking forward to printing one for my Sanni.

**Comment by sanni** [comment]
Awesome ❤️ 

If you want you can also do a pull request and upload it into a sub-folder under hardware.

**Reply by KytorIndustries** [reply]
@sanni I have an updated version of the shell I am going to release soon. I could do a pull request so it can be added. Before I do, have any of the overall geometry or externally accessible component locations changed in the new PCB revisions?

**Reply by Ancyker** [reply]
@KytorIndustries With Vselect installed, there is no voltage switch:
![image](https://github.com/sanni/cartreader/assets/7105333/dd97295c-ca08-4e37-9bf7-46bef066e536)

The logo on the board changed slightly, only matters if creating a hole to be able to see it:
![oscr-hw5-pcb-logo-rev5](https://github.com/sanni/cartreader/assets/7105333/1756760e-c03e-4dbc-9259-f202275e700f)

There is also the RTC battery holder.

Overall, the board is the same:
![oscr-hw5-pcb-kit-1-rev5](https://github.com/sanni/cartreader/assets/7105333/3af8f4ec-d439-42bb-978a-0699f4dcdf9a)

I hopefully will be committing REV5 today, about to assemble/test it and if it works I will be.

**Reply by KytorIndustries** [reply]
Thank you. 

@breyell did give me a heads up already on the vselect feature, very neat. I'll probably grab one ready-made from Starshade. I can make a bottom shell variant with a voltage switch window delete.

I ended up deleting the logo window on the bottom.

**Reply by Ancyker** [reply]
@KytorIndustries Might want to be mindful of the RTC battery holder, it even interferes with the existing frame, so someone will need to retool that too. I got a 3D printer recently but only the printer, gotta get some material XD

![2023-07-16 00 03 18](https://github.com/sanni/cartreader/assets/7105333/ec127eb0-6601-4848-99f8-67849a30d477)

**Reply by KytorIndustries** [reply]
@Ancyker I assume your photo is of the "Simple Frame"? If so, that design features standoffs with a 2mm gap between the board bottom and the base.

![image](https://github.com/sanni/cartreader/assets/75507599/92eb79a9-68af-44b1-8500-73bb5a191935)

The same applies to the "Main PCB Stand" designs.

![image](https://github.com/sanni/cartreader/assets/75507599/5cd3772e-8197-4d6d-bc06-0d857de63695)

The Fully Enclosed OSCR Shell (FEOS) is 2.5mm:

![image](https://github.com/sanni/cartreader/assets/75507599/93564576-9694-41fb-ab0e-dd8991478469)

The HW5 REV5 schematic lists a CR1616 battery.
[Here is the cut-sheet](https://www.mouser.com/datasheet/2/209/EaglePlasticDevices_7.17.2017.EPD-200984-1171085.pdf) for a PCB mount CR1616 holder. It lists a 4mm total height. Yours pictured is a slightly different style, in at least the spring cut-outs.

I have a half dozen other styles of battery holders on hand, but not that one. Do you have a pair of vernier to confirm this dimension?

On the other hand, if you directly solder a tabbed CR1616, you will not have any issue with my FEOS design. The battery is about 2.05mm tall (after subtracting 0.15mm thick plastic; don't want to measure without for obvious reasons). Perhaps the design intent is to use a tabbed battery, and not a removable battery holder? The RTC battery should not need regular intervention.

![image](https://github.com/sanni/cartreader/assets/75507599/48317854-d660-4b80-ab4d-65f0ea83166d)

**Comment by Ancyker** [comment]
![2023-07-16 10 47 55](https://github.com/sanni/cartreader/assets/7105333/3c160e5c-d02c-482e-a629-b591629eae88)

**Comment by Ancyker** [comment]
Here is the datasheet (3012): https://www.keyelco.com/userAssets/file/M65p9.pdf

**Reply by KytorIndustries** [reply]
That is peculiar the measurement is that far off. The data sheet lists 4.0mm for the 3012 holder.

**Reply by Ancyker** [reply]
The PCB is 1.6mm

**Reply by KytorIndustries** [reply]
Did not notice the PCB in the photo. I see that now. Perfect. So we can assume 4mm plus some margin.

**Comment by KytorIndustries** [comment]
I will add a window for battery removal. Unless we think tabbed (soldered on) CR1616 is going to be the standard?

![image](https://github.com/sanni/cartreader/assets/75507599/0ab69526-a53e-4b5f-a400-962cc45b3fd2)

**Reply by Ancyker** [reply]
I don't really like the idea of a tabbed battery, nor does the current footprint support one.

**Reply by Ancyker** [reply]
Both pads on either side are positive. The center pad is negative/ground.

![image](https://github.com/sanni/cartreader/assets/7105333/8eb2801c-6bdb-4cef-891d-d1dd282bd5e1)

**Reply by KytorIndustries** [reply]
Yep, understood. The only real viable option to accommodate without significant rework is adding a window. The positive consequence is that will allow battery access for ease of maintenance.

**Reply by Ancyker** [reply]
Just note it's designed to slide in and out in this direction:

![image](https://github.com/sanni/cartreader/assets/7105333/9720048c-186d-49cd-b80f-c28f2cbf75ec)

**Comment by KytorIndustries** [comment]
Needs to be validated on actual hardware, but this should do the trick. Includes a bumper to prevent dropping the battery into the pit of death. I can also make a little cover with detents to snap in.

![image](https://github.com/sanni/cartreader/assets/75507599/2f9852a6-b7b1-4054-808d-b0fdd44fd550)

![image](https://github.com/sanni/cartreader/assets/75507599/817031d0-e390-444c-99b5-f00e33be7694)

**Reply by Ancyker** [reply]
Lol @ pit of death. Apt description too, as dropping it in there it could short the battery out and make it catch on fire.

A snap-in cover would be cool. Do you know if this case is resin-friendly? The basic frame isn't because of the heat-set inserts.

Also, have you thought of eliminating the visible screws on the top?

**Reply by KytorIndustries** [reply]
This also uses some heatserts.

The top cover has been reduced to three visible screws. Two on the sides for the faceplate.

![IMG_7350](https://github.com/sanni/cartreader/assets/75507599/f84b494d-4390-42dc-a798-f75299d3cb89)

**Reply by Ancyker** [reply]
I was thinking of ways to eliminate the visible screws. I picked these up:

![image](https://github.com/sanni/cartreader/assets/7105333/50e3f368-4f4d-4302-afb5-fc1ba5d47eb1)

Had the idea to glue them into the frame. Then use standoffs with a male connector on one end and a female on the other. Glue the nuts into the top of the housing, then put the PCB in, then the standoffs into the nut that was glued in. Then drop in the main PCB and the bottom part of the frame. Screw the frame and PCB down into the female end of the standoffs. No visible screws then.

**Reply by KytorIndustries** [reply]
I personally avoid the use of glue in my designs, but the shell would be simple to remix for various alternate fastener methods.

**Reply by Ancyker** [reply]
I'd rather glue in something durable than use something self-tapping as self-tapping threads will wear out super quickly if you take it apart a lot (and I do).

**Reply by Ancyker** [reply]
I suppose you could make them slide into place and be held in that way? sort of like how the battery holder works

**Reply by KytorIndustries** [reply]
That is why I only specify self-tapping in areas with infrequent removal (like the LCD), whereas all other fasteners are metal hardware.

**Reply by Ancyker** [reply]
Fair enough. I'm excited to print one eventually, I want to make a clear one. I miss stuff with clear housings :3

**Comment by iceyman** [comment]
Has https://www.printables.com/model/409926 been updated?

**Reply by KytorIndustries** [reply]
Not yet, Rev 2 is still in process.

**Reply by ccfman2004** [reply]
Can't wait to see it.  Keep up the good work. I can't CAD to save my life.

**Comment by KytorIndustries** [comment]
Do you know what's better than only 2 visible screws? Zero visible screws. Nice and secure clip-in action.

![IMG-7487](https://github.com/sanni/cartreader/assets/75507599/b83f496a-49b1-42a5-8b9f-b4b3fc6ce5af)

**Reply by ccfman2004** [reply]
Nice.

**Reply by dtomcat** [reply]
🤤

**Reply by Ancyker** [reply]
Nice. I'm curious, how long does it take you to print that?

**Reply by KytorIndustries** [reply]
About 5 hours on the X1C, settings slowed down, and 10.5 hours on the MK3S+.

**Reply by iceyman** [reply]
Pla or petg? I have done mine in pla going to try petg next

**Reply by KytorIndustries** [reply]
PLA. PETG is going to be a bit more work because supports should be used in a few areas (e.g. GBA, N64 openings). Organic/tree supports should help.

**Reply by Ancyker** [reply]
Is it 100% snap together? No screws at all?

**Reply by KytorIndustries** [reply]
No visible screws from the top and sides. There are still screws on the bottom.

**Reply by Ancyker** [reply]
Are they just directly fastened or do they need inserts? Just wondering if I could print it on my printer (resin).

![2023-07-25 18 44 30c](https://github.com/sanni/cartreader/assets/7105333/c94e122f-5c23-4336-8773-5bbd7c4ea7d5)

**Reply by KytorIndustries** [reply]
Just two heat set nuts into the faceplate. I'm not very familiar with resin printing so I'm not sure.

**Reply by Ancyker** [reply]
Resin doesn't melt so heat setting doesn't work. :3

**Reply by Ancyker** [reply]
For FDM, you'd normally do say, something like this:
![image](https://github.com/sanni/cartreader/assets/7105333/a8c67815-7bcf-49fc-8787-2fb32a7d4095)

To be compatible with both FDM and SLA, you could do something like this:
![image](https://github.com/sanni/cartreader/assets/7105333/ae29888a-482b-4653-a1e8-6587197210a7)

There are other ways to do it as well, such as having inserts for the nuts, or just making the hole too small so the screws make their own threads. Screwing into 3D prints directly is actually pretty strong, the issue is the material doesn't take to having the screws removed and reinserted a bunch. As a once-off, it's actually equal to (and sometimes stronger than) the heat-set inserts.

Edit: Also, please pardon my MS Paint drawings 😅🤣

**Reply by Ancyker** [reply]
Here are some videos on inserts/fastening methods, the first one is about SLA and the other 2 are FDM:

https://www.youtube.com/watch?v=4nbvU2kGzxc
https://www.youtube.com/watch?v=iR6OBlSzp7I
https://www.youtube.com/watch?v=G-UF4tv3Hvc

**Comment by KytorIndustries** [comment]
If anyone in the contiguous USA is actively building a HW5 REV5 OSCR with the RTC, I would like to send you a sample of the new shell to verify fit up. DM me on Twitter @KytorIndustries if you're interested.

Edit: A guinea pig has been chosen.

**Reply by ccfman2004** [reply]
I’m just waiting on the PCB and components. Just send me the stl as I have a 3D printer. I built a Tridex that’s going to ERRF this year. 

Twitter @ccfman2004

**Comment by KytorIndustries** [comment]
I forgot to follow up here and let folks know that rev 2 is released on Printables:
https://www.printables.com/model/409926

This is a major overhaul including many hundreds of changes, too many to list to make the shell more durable, better looking, easier to install, vSelect option, RTC battery access, multi-color options, and have no visible hardware on the top and sides. Now also including a dust cover.
